include/openssl/core_names.h

```diff
@@ -34,6 +34,31 @@ extern "C" {
  */
 #define OSSL_PROV_PARAM_BUILDINFO   "buildinfo"
 
+/*
+ * The modules filename including the path (Used for FIPS Self Testing)
+ * Type: OSSL_PARAM_UTF8_STRING
+ */
+#define OSSL_PROV_PARAM_MODULE_FILENAME "module-filename"
+/*
+ * The calculated MAC of the module file (Used for FIPS Self Testing)
+ * Type: OSSL_PARAM_UTF8_STRING
+ */
```

Comment:
Is this meant to be the file name of the .so/.dll/.exe (it's .exe on VMS)?  I'm not keen on that, and would rather see something that indicates the file name of the checksum file to be read.  That avoids all the hassle you're going to have with editing the file name.

Also, do note that the `ossl_provider_` API already knows the module file name.  If this is a needed parameter after all, this should probably become a parameter that `core_get_params` handles natively...

Reply:
Inside the fips module it needs to perform a HMACSHA256 on the fips module image, so I need the filename from somewhere..
OK will changes.. thanks..

Reply:
> Inside the fips module it needs to perform a HMACSHA256 on the fips module image, so I need the filename from somewhere..

Ah!  I keep forgetting that...

Reply:
updated